staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -489,6 +525,9 @@ func (td *typeDiscoverer) discoverStruct(thisNode *typeNode, fldPath *field.Path
 				childType: childType,
 				node:      node,
 			}
+			// Extract +listMapKeys for correlating object and oldObject
+			// during update validation.
+			node.listMapKeys = extractListMapKeys(memb)
```

Comment:
Oh good question.  I was thinking of update validation as being part of the machinery, and not a plugin, and since it relies on knowing the keys of listType=map, I just went ahead and hard coded a understanding that into validation-gen.

But if we about the listType=map plugin (which we will absolutely need), then this gets more interesting.  The plugin itself would be responsible for:
- Ensuring that a slice has unique keys according to the +listMapKey=<key> tags.
- Somehow deal with cases like container Envs where duplicates are allowed

Maybe when we add listType=map plugin support we can consolidate the tag access code a bit?  I'm trying to think of other ways to organize this better but haven't though of anything super helpful.

Reply:
Yeah, when I started reading this I was thinking of +listMapKey as looking for dup keys, but this doesn't do that at all, just the update tracking.